### ISSUE TITLE
Generating output images larger than 256x256

### ISSUE THREAD

**Issue by nonoesp** [affinelayer/pix2pix-tensorflow#22]
Is there a way to specify the size of the output images to be anything other than 256x256?

Thanks!

**Comment by christopherhesse** [comment]
I haven't tried this, but probably the best way to do it is to add 2 layers in the generator (one on encode side and one on decode side) and then add one in the discriminator to get to 512x512, and the same to get to 1024x1024.

**Comment by ksteinfe** [comment]
I suppose that simply changing the CROP_SIZE constant (line 51) and the --scale_size parser argument (line 37) doesn't do the trick? We just tried this, and things appear to be working still, but perhaps we're missing something important? 

I can find where the layers are defined in both the generator (lines 339 / 357) and discriminator (407), but am new to tensorflow, and I'm not confident that I can add the needed layers without breaking things. To move to 512x512, would I also need to adjust the number of generator/discriminator filters in first conv layer, as defined on lines 35/36?

Has anyone successfully done this?

**Comment by julien2512** [comment]
Yes it works ! I tryed 1024x1024 pixels, with only one more conv layer.

**Comment by julien2512** [comment]
But as the title change, you may want to have an input 256*256 with a larger output ? You may experience problems with the "bypass" then, you should avoid in the last discriminator conv layer.

**Comment by julien2512** [comment]
Hi,

I tryed 1024x1024 successfully, despite of the Time it Takes !

You may want to change the number of conv layers indeed or Their respective
size (as 40 might vive poor results).

I am looking for other improvements too !

Regards,
Julien

**Comment by amlarraz** [comment]
@julien2512 can you explain or share your changes in the code?? I'm trying to add the layers but I'm new in tf and I don't know where is my error...':) thanks in advance!

**Comment by julien2512** [comment]
@amlarraz  You don't really need to change the code in order to make it. It is only about how deep you want to reduce your data dimensions. I still have a lot to learn about it.

Eventhough, I tryed myself to add a layer in the two "layer_specs" array like this (in pix2pix.py), and it works :
`def create_generator(generator_inputs, generator_outputs_channels):
    layers = []

    # encoder_1: [batch, 256, 256, in_channels] => [batch, 128, 128, ngf]
    with tf.variable_scope("encoder_1"):
        output = conv(generator_inputs, a.ngf, stride=2)
        layers.append(output)

    layer_specs = [
        a.ngf * 2, # one more layer
        a.ngf * 2, # 
        a.ngf * 4, # 
        a.ngf * 8, # 
        a.ngf * 8, # 
        a.ngf * 8, # 
        a.ngf * 8, # 
        a.ngf * 8, # 
    ]

    for out_channels in layer_specs:
        with tf.variable_scope("encoder_%d" % (len(layers) + 1)):
            rectified = lrelu(layers[-1], 0.2)
            # [batch, in_height, in_width, in_channels] => [batch, in_height/2, in_width/2, out_channels]
            convolved = conv(rectified, out_channels, stride=2)
            output = batchnorm(convolved)
            layers.append(output)

    layer_specs = [
        (a.ngf * 8, 0.5),   # 
        (a.ngf * 8, 0.5),   # 
        (a.ngf * 8, 0.5),   # 
        (a.ngf * 8, 0.0),   # 
        (a.ngf * 4, 0.0),   # 
        (a.ngf * 2, 0.0),   # 
        (a.ngf * 2, 0.0),   # one more layer
        (a.ngf, 0.0),       # 
    ]

    ...
`

You can change the discrimator graph the same way. But it's the data that will lead you to change both generator and discriminator. Maybe it is necessary to try many models before !

**Comment by amlarraz** [comment]
Nice!! Thanks for the fast response!! I'm going to try and I'll tell you my advances, thanks @julien2512 !

**Comment by orenyomtov** [comment]
Hi @julien2512  @amlarraz ,

I'm trying to get it working with an input and output layer size of 512x512 grayscale, but didn't figure out how to do so yet.

Did any of you succeed in doing this?

**Comment by julien2512** [comment]
Hi @orenyomtov,
I did not try to do this, but it might work as this code trying to guess the output color channels from your image samples :
`
out_channels = int(targets.get_shape()[-1])
outputs = create_generator(inputs, out_channels)
`
Let us know how it's behave !

**Comment by Ji-qing** [comment]
@julien2512 
Hello,
I am a totally newbie in deep learning and also working with image data.You give the code in generator, I do not know how to change the code in discriminator. I want to have an input/output 1024*1024. Can you share your changes in discriminator? Thanks!!

**Comment by Ji-qing** [comment]
@amlarraz 
Hello,
 I wonder if you  have an input/output 1024*1024 successfully. Can you share your changes in discriminator? Thanks!!

**Comment by Ji-qing** [comment]
@julien2512 
Hello,
I am a totally newbie in deep learning and also working with image data.You give the code in generator, I do not know how to change the code in discriminator. I want to have an input/output 1024*1024. Can you share your changes in discriminator? Thanks!!

**Comment by julien2512** [comment]
Hi @Ji-qing,

You can add 2 more layers in :

n_layers = 3

or you can leave it as it is ! Discriminator is about how to compare images
together. With a 1024 image, it will try to compare something like a 120
characteristic wide image. With 5 layers it will end to 30, as it is with a
256 image.

Regards, Julien.


Le 17 juil. 2017 14:16, "Ji-qing" <notifications@github.com> a écrit :

@julien2512 <https://github.com/julien2512>
Hello,
I am a totally newbie in deep learning and also working with image data.You
give the code in generator, I do not know how to change the code in
discriminator. I want to have an input/output 1024*1024. Can you share your
changes in discriminator? Thanks!!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/affinelayer/pix2pix-tensorflow/issues/22#issuecomment-315739329>,
or mute the thread
